plan-for-extension-exercise/src/planForExtensionExercise/TotalAreaCalculator.java

```diff
@@ -0,0 +1,25 @@
+package planForExtensionExercise;
+
+public class TotalAreaCalculator extends AreaCalculator {
+	public static double calculateArea(Shape[] shapes) {
+		double area = 0.0;
+		if (shapes == null) {
+			return area;
```

Comment:
We could certainly have the method throw an InvalidParameterException if
the shapes array is null, with a message informing the user of this. This
would involve changing the method signature (as you indicated) to include
"throws InvalidParameterException" and upon checking the parameter in the
code, throw that exception with an appropriate message.

On Thu, Feb 5, 2015 at 4:18 PM, Sergey A. Razin <notifications@github.com>
wrote:

> In
> plan-for-extension-exercise/src/planForExtensionExercise/TotalAreaCalculator.java
> <https://github.com/siostechcorp/exercises/pull/4#discussion_r24200402>:
>
> > @@ -0,0 +1,25 @@
> > +package planForExtensionExercise;
> > +
> > +public class TotalAreaCalculator extends AreaCalculator {
> > +	public static double calculateArea(Shape[] shapes) {
> > +		double area = 0.0;
> > +		if (shapes == null) {
> > +			return area;
>
> That's certainly valid, however could be misleading. Is area of a null 0?
> Maybe yes...what if you truly want the user to know that what was passed in
> is in fact an unexpected parameter (like null). What would you do without
> changing the return type, however (hint) it may require a method signature
> to be changed, and how?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/siostechcorp/exercises/pull/4/files#r24200402>.
>